rllib/algorithms/dqn/dqn_rainbow_catalog.py

```diff
@@ -0,0 +1,191 @@
+# __sphinx_doc_begin__
+import gymnasium as gym
+
+from ray.rllib.algorithms.dqn.dqn_rainbow_noisy_net_configs import (
+    NoisyMLPEncoderConfig,
+    NoisyMLPHeadConfig,
+)
+from ray.rllib.core.models.catalog import Catalog
+from ray.rllib.core.models.base import Model
+from ray.rllib.core.models.configs import MLPHeadConfig
+from ray.rllib.models.torch.torch_distributions import TorchCategorical
+from ray.rllib.utils.annotations import (
+    ExperimentalAPI,
+    override,
+    OverrideToImplementCustomLogic,
+)
+
+
+@ExperimentalAPI
+class DQNRainbowCatalog(Catalog):
+    """The catalog class used to build models for DQN Rainbow.
+
+    `DQNRainbowCatalog` provides the following models:
+        - Encoder: The encoder used to encode the observations.
+        - Target_Encoder: The encoder used to encode the observations
+            for the target network.
+        - Af Head: Either the head of the advantage stream, if a dueling
+            architecture is used or the head of the Q-function.
+        - Vf Head (optional): The head of the value function in case a
+            dueling architecture is chosen.
+
+    All networks can include noisy layers, if `noisy` is `True`.
+
+    Any custom head can be built by overridng the `build_af_head()` and
+    `build_vf_head()`. Alternatively, the `AfHeadConfig` or `VfHeadConfig`
+    can be overridden to build custom logic during `RLModule` runtime.
+
+    All heads can optionally use distributional learning. In this case the
+    number of output neurons corresponds to the number of actions times the
+    number of support atoms of the discrete distribution.
+    """
+
+    @override(Catalog)
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        model_config_dict: dict,
+        view_requirements: dict = None,
```

Comment:
Actually, now that we don't support view_requirements anymore, we should add an assert that this is None and then error with a meaningful message (saying something like: "instead, use the new ConnectorV2 API to pick whatever information you need from the running episodes").